openpype/modules/deadline/plugins/publish/submit_houdini_render_deadline.py

```diff
@@ -262,6 +268,11 @@ def get_plugin_info(self, job_type=None):
                 plugin_info = VrayRenderPluginInfo(
                     InputFilename=instance.data["ifdFile"],
                 )
+            elif family == "redshift_rop":
+                plugin_info = RedshiftRenderPluginInfo(
+                    SceneFile=instance.data["ifdFile"],
+                    Version=os.getenv("REDSHIFT_VERSION", "3.5.22"),
```

Comment:
I personally find the version fallback odd here. It's much clearer maybe to just require the var to be set so it's clear that it's used?

```suggestion
                    Version=os.environ["REDSHIFT_VERSION"],
```
That is, if that's a variable that's typically set for Redshift. _**It doesn't appear to be set on our end.**_ in the tool settings at least. So I'm not sure it's a required variable.

---

Or we get the Redshift version directly via a Python call to the Redshift API or whatever? However, I don't quickly see something we can call to do so.

Reply:
I couldn't find a way to get Redshift version from python at all. Version is set in `PluginInfo`  and it plays a role if you have multiple executables set in Redshift Deadline plugin. That one is oddly behaving differently - they expect that if you have more than one version of Redshift, that you add more `[Redshift_Executable_*]` entries in **Redshift.param** in the plugin itself and the version plays a role at that point. I guess it shouldn't set it if that env var isn't set though and so it will support multiple versions configured and single one too.

Reply:
> That one is oddly behaving differently - they expect that if you have more than one version of Redshift, that you add more [Redshift_Executable_*] entries in Redshift.param in the plugin itself and the version plays a role at that point.

This sounds very close to the issue with Houdini versions that @MustafaJafar battled some time ago - which could be resolved by OpenPype/AYON setting `PATH` for Houdini bin folder and just specifying `hython.exe` without path in Deadline without the need to get Houdini version specific for the paths specified in Deadline itself because OpenPype would be initializing the correct version. The same could be done for Redshift tools, etc.

Anyway, I couldn't find another way to get the version quickly either. But isn't this issue also present for the arnold and v-ray submissions? Those do not seem to care about a version getting submitted along?

Reply:
I guess with arnold, vray and redshift it works by pointing path to specific version. I've looked to default paths set by deadline there and it is something like:
```
/opt/solidangle/mtoa/2018/bin/kick;/opt/solidangle/mtoa/2017/bin/kick
```
for arnold - that's even more confusing as they expect to first find the "newer" version and if that fails use the older one? I'll have to look into the plugin code itself if it actually supports version differentiation. Same goes for vray. Redshift on the other hand supports this trick with `Version`

Reply:
Ah yes...

```python
return self.GetRenderExecutable("Arnold_RenderExecutable_" + version , "Arnold " + version)
```

Reply:
Yup, most deadline plugins allow to specify a version.

The trick with `PATH` like I described then works by specifying just the executable in that plug-in setting so that e.g. Houdini 19.5 submission can run for 19.5.585 or 19.5.805 depending on the actual application set up in OpenPype/AYON in that project, etc.

Anyway, not sure what the best way in this case is to specify redshift version etc. I feel like we should at least avoid the random fallback to a hardcoded version. And if we go for `REDSHIFT_VERSION` env var we should document that _that_ is what studios should be setting as env var for Redshift since it's not there by default?

Reply:
So consistency:

**Redshift**
```python
version = self.GetPluginInfoEntryWithDefault( "Version", "1" )
```

**Arnold**
```python
version = self.GetPluginInfoEntryWithDefault( "Version", "Release" )
```

**Vray**
```python
...
vrayExe = self.GetRenderExecutable( "VRay_RenderExecutable", "V-Ray" )
self.vrayVersion = self.getVrayVersion( vrayExe )
return vrayExe
```
(but `self.vrayVersion` isn't used anywhere)

Reply:
> And if we go for `REDSHIFT_VERSION` env var we should document that _that_ is what studios should be setting as env var for Redshift since it's not there by default?

I would go with the `*_VERSION` - add it to defaults and document it. We are already doing it with `MTOA_VERSION` (although that tool itself should be clearly split between different flavours of anrold (mtoa, htoa, standalone, ...)) We are setting `MAYA_VERSION` to be used in paths, so I guess there is not harm to do it for Redshift too. I am not sure what to do with Vray in this regard, maybe that trick with `PATH` - but it is not explicit enough for my taste (one wrong setting or different order in `PATH` and something else is picked, hard to debug)

Reply:
Ugh - oh Deadline.

Anyway, we might want to still document the `REDSHIFT_VERSION` usage in the codebase for admins to understand that _that_ is what they can set. We might even want to log a warning if it's not set?

Reply:
One remaining thing - say you specify version `3.5.22` in `REDSHIFT_VERSION`. What's the equivalent "parm" for the `Redshift.param` file? I suspect it won't support `.` (period)  in the name there?

Reply:
would you guys excuse my noob question ? 

when I have installed redshift on my PC. I didn't find any versions specified in the bin folder
![image](https://github.com/ynput/OpenPype/assets/20871534/2a641045-c7c5-465c-ab95-37e7e7e3665b)

also, the default settings don't say anything about redshift version 
![image](https://github.com/ynput/OpenPype/assets/20871534/6e56de4c-9e14-4e2b-b21b-94a1ba7b8001)

has that happened because I didn't change default installation path ?

Reply:
> when I have installed redshift on my PC. I didn't find any versions specified in the bin folder

Correct, the local install is always the local install. However, we run a Redshift tool per project (sometimes even per shot) to define what Redshift version is used. There are no local installs on our end.

> also, the default settings don't say anything about redshift version

It does, it refers to it as version 1 as the label says. Which is the default version used if no version is passed along.

> It doesn't support `.` in the name between `[Redshift_Executable_***]`

That would mean the `REDSHIFT_VERSION` variable used @antirotor would need to be adhere to that too making this quite odd for any developer that `3.5.22` needs to be `3_5_22`. I think we're currently best off currently removing the whole version environment variable part of this PR and not passing it along to match the other implementations. Especially since Deadline Redshift plug-in by default exposes only one entry anyway.

If studio exposes Redshift to OpenPype as "tools" or application environment variables and they add `PATH` of `Redshift` then they can just specifcy only this in the deadline config:
```
redshiftCmdLine.exe
redshiftCmdLine
```
Which will then pick up the relevant version for OpenPype/AYON automatically.

Reply:
> One remaining thing - say you specify version `3.5.22` in `REDSHIFT_VERSION`. What's the equivalent "parm" for the `Redshift.param` file? I suspect it won't support `.` (period) in the name there?

I am surprised myself, but it does support it (so no conversion is needed) - it really just append string.

Reply:
Oh and equivalent "parm" string is `[Redshift_Executable_3.5.22]`.